src/cpl/share_esmf/PrigentRoughnessStreamType.F90

```diff
@@ -0,0 +1,356 @@
+
+  !==============================================================================
+  logical function IsStreamInit(this)
+    !
+    ! !DESCRIPTION:
+    ! Return true if the streams have been initialized
+    !
+    ! !USES:
+    !
+    ! !ARGUMENTS:
+    implicit none
+    class(prigent_roughness_stream_type) :: this
+    !
+    character(len=*), parameter :: subname = 'PrigentRoughnessStream::IsStreamInit'
+    !
+    if ( .not. NMLRead )then
+       call endrun(msg=subname//' ERROR Namelist has NOT been read first, call Init before this')
+    end if
+    if ( InitDone )then
+       IsStreamInit = .true.
+    else
+       IsStreamInit = .false.
+    end if
+  end function IsStreamInit
+
+  !==============================================================================
+  subroutine Clean(this)
+   !
+   ! Deallocate and clean the object
+   !
+   ! Uses:
+   !
+   ! arguments
+   implicit none
+   class(prigent_roughness_stream_type) :: this
+   !
+   ! local variables
+   !-----------------------------------------------------------------------
+   deallocate(this%prigent_rghn)
+   this%prigent_rghn => NULL()
+   InitDone = .false.
+
+  end subroutine Clean
+
+  !==============================================================================
+  subroutine InitAllocate(this, bounds)
+    !
+    ! !DESCRIPTION:
+    ! Allocate module variables and data structures
+    !
+    ! !USES:
+    use shr_infnan_mod, only: nan => shr_infnan_nan, assignment(=)
+    !
+    ! !ARGUMENTS:
+    implicit none
+    class(prigent_roughness_stream_type) :: this
+    type(bounds_type), intent(in) :: bounds
+    !
+    ! !LOCAL VARIABLES:
+    integer  :: begg, endg
+    !---------------------------------------------------------------------
+
+    begg = bounds%begg; endg = bounds%endg
+
+    if ( this%useStreams() )then
+       allocate(this%prigent_rghn(begg:endg))
+    else
+       allocate(this%prigent_rghn(0))
+    end if
+    this%prigent_rghn(:) = nan
+
+  end subroutine InitAllocate
+
+  !==============================================================================
+  subroutine ReadNML(this, bounds, NLFilename)
+   !
+   ! Read the namelist data stream information.
+   !
+   ! Uses:
+   use shr_nl_mod       , only : shr_nl_find_group_name
+   use shr_log_mod      , only : errMsg => shr_log_errMsg
+   use shr_mpi_mod      , only : shr_mpi_bcast
+   !
+   ! arguments
+   implicit none
+   class(streamcontrol_type)     :: this
+   type(bounds_type), intent(in) :: bounds
+   character(len=*),  intent(in) :: NLFilename   ! Namelist filename
+   !
+   ! local variables
+   integer            :: nu_nml    ! unit for namelist file
+   integer            :: nml_error ! namelist i/o error flag
+   logical            :: use_prigent_roughness = .true.
+   character(len=CL)  :: stream_fldFileName_prigentroughness = ' '
+   character(len=CL)  :: stream_meshfile_prigentroughness = ' '
+   character(len=CL)  :: prigentroughnessmapalgo = 'bilinear'
```

Comment:
Make an issue that this is hardcoded to bilinear. And it should be made into a namelist item handled by build-namelist, and also change the name to:

prigent_roughness_mapalgo

Reply:
#2680